### PR TITLE
#161967098 Tags present in all relevant views

### PR DESCRIPTION
**Pull request by Darius-Ndubi**
**What does this PR do?**
- Adds a tags on the discover page
- Filters articles as per a tag on another article

**Description of tasks to be completed?** 
- Allows users to find other articles with tags on another article.

**How should this be manually tested?**
- Clone the repo and cd into the project root folder
- Switch to the relevant branch `git checkout ft-tags-present-relevant-views-161967098`
- Run `npm install` to update your packages.
- Start the server with `npm start`
- In your favorite browser, navigate to `http://localhost:3000`

**What are the relevant pivotal tracker stories?**
[#161967098](https://www.pivotaltracker.com/story/show/161967098)

##### A tag on article
screenshot 2019-01-25 at 11 53 11

##### Discover Page
screenshot 2019-01-25 at 11 53 42

##### Mobile view
screenshot 2019-01-25 at 16 36 44

** Checklist **
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by justmesam** [comment]
Look at the merge conflicts, else LGTM

File: src/components/containers/discoverPage.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { Item, Container } from "semantic-ui-react";
+import PropTypes from "prop-types";
+import { bindActionCreators } from "redux";
+import Article from "../views/article2";
+import * as tagSearching from "../../actions/tagSearchActions";
```
Comment (bettblake08) [inline src/components/containers/discoverPage.js]: I think it's better to have named it tagSearchingActions.

File: src/components/views/article2.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+};
+
+Article.propTypes = {
+  article: PropTypes.object
+};
+
+Article.defaultProps = {
+  article: {
+    id: 1,
+    slug: "mouth-letter-commercial-happen-allow",
```
Comment (bettblake08) [inline src/components/views/article2.js]: Set these string defaults to "".

File: src/components/views/articles/SingleArticle.js [hunk 3]
```diff
@@ -10,6 +12,7 @@ import { Link } from "react-router-dom";
 
 import * as actionGenerators from "../../../actions/articlesActions";
 import HeaderComponent from "../../containers/headers/index";
+import * as tagSearching from "../../../actions/tagSearchActions";
```
Comment (bettblake08) [inline src/components/views/articles/SingleArticle.js]: I think it's better to have named it tagSearchingActions.

File: src/reducers/index.js [hunk 4]
```diff
@@ -5,12 +5,14 @@ import registrationReducer from "./registrationReducer";
...
 
 export default history => combineReducers({
   router: connectRouter(history),
   registrationReducer,
   loginReducer,
   userSettingsReducer,
   profileReducer,
-  article
+  article,
+  tagReducer
```
Comment (bettblake08) [inline src/reducers/index.js]: We have been told to remove the suffix Reducer from these keys.

File: src/store/initialState.js [hunk 5]
```diff
@@ -25,5 +25,8 @@ export default {
     singleArticle: {
 
     }
+  },
+  tagsearched_articles: {
```
Comment (bettblake08) [inline src/store/initialState.js]: tagsearched_articles does not match any key in the reducers index. Is it being used?

**Review by bettblake08 — CHANGES_REQUESTED** [review]

File: src/components/containers/discoverPage.js (hunk 1 above)
Comment (Darius-Ndubi) [reply src/components/containers/discoverPage.js]: @bettblake08 Let me rename it to that.

File: src/components/views/articles/SingleArticle.js (hunk 3 above)
Comment (Darius-Ndubi) [reply src/components/views/articles/SingleArticle.js]: @bettblake08 let me rename it to fit its function

File: src/reducers/index.js (hunk 4 above)
Comment (Darius-Ndubi) [reply src/reducers/index.js]: @bettblake08 thanks for the reminder let me work on it.

File: src/store/initialState.js (hunk 5 above)
Comment (Darius-Ndubi) [reply src/store/initialState.js]: Yes it is. Let me look into that.

**Review by bettblake08 — APPROVED** [review]

**Review by davishooly — APPROVED** [review]

**Review by denomash — APPROVED** [review]
👍

**Pull request merged**